docs/OV_Runtime_UG/auto_device_selection.md

```diff
@@ -1,332 +1,410 @@
-```
-
-Alternatively, you can combine all the individual device settings into single config file and load it, allowing the Auto-device plugin to parse and apply it to the right devices. See the code example here:
+For C++ API
+@sphinxdirective
+.. code-block:: sh
 
-```python
-from openvino.inference_engine import IECore
+   ov::runtime::Core::get_available_devices() (see Hello Query Device C++ Sample)
+@endsphinxdirective
 
-# Init the Inference Engine Core
-ie = IECore()
+For Python API
+@sphinxdirective
+.. code-block:: sh
 
-# Read a network in IR or ONNX format
-net = ie.read_network(model=path_to_model)
+   openvino.runtime.Core.available_devices (see Hello Query Device Python Sample)
+@endsphinxdirective
 
-full_config = {}
 
-# Load the network to the AUTO device
-exec_net = ie.load_network(network=net, device_name="AUTO", config=full_config)
-```
+### Performance Hints
+The `ov::hint` property enables you to specify a performance mode for the plugin to be more efficient for particular use cases.
 
-### Using the Auto-Device with OpenVINO Samples and Benchmark App
+#### ov::hint::PerformanceMode::THROUGHPUT
+This mode prioritizes high throughput, balancing between latency and power. It is best suited for tasks involving multiple jobs, like inference of video feeds or large numbers of images.
 
-Note that every OpenVINO sample or application that supports the "-d" (which stands for "device") command-line option transparently accepts the Auto-device. The Benchmark Application is the best example of the optimal usage of the Auto-device. You do not need to set the number of requests and CPU threads, as the application provides optimal out-of-the-box performance. Below is the example command-line to evaluate AUTO performance with that:
+#### ov::hint::PerformanceMode::LATENCY
+This mode prioritizes low latency, providing short response time for each inference job. It performs best for tasks where inference is required for a single input image, like a medical analysis of an ultrasound scan image. It also fits the tasks of real-time or nearly real-time applications, such as an industrial robot's response to actions in its environment or obstacle avoidance for autonomous vehicles.
+Note that currently the `ov::hint` property is supported by CPU and GPU devices only.
 
+To enable Performance Hints for your application, use the following code: 
 @sphinxdirective
-.. tab:: Package, Docker, open-source installation
-
-   .. code-block:: sh
-
-      ./benchmark_app.py –d AUTO –m <model>
+.. tab:: C++ API
+
+   .. code-block:: cpp
+
+      ov::Core core;
+
+      // Read a network in IR, PaddlePaddle, or ONNX format:
+      std::shared_ptr<ov::Model> model = core.read_model("sample.xml");      
+      
+      // Load a network to AUTO with Performance Hints enabled:
+      // To use the “throughput” mode:
+      ov::CompiledModel compiled_model = core.compile_model(model, "AUTO:GPU,CPU", {{ov::hint::performance_mode.name(), "THROUGHPUT"}});
```

Comment:
`{{ov::hint::performance_mode.name(), "THROUGHPUT"}}` => `ov::hint::performance_mode(ov::hint::PerformanceMode::THROUGHPUT)` and the same for latency below